.github/workflows/ci.yml

```diff
@@ -0,0 +1,94 @@
+name: Scala CI
+
+on:
+  push:
+    branches: [main]
+    tags: ["v*"]
+  pull_request:
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Scala
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: adopt@1.11
+
+      - name: Checking formating
+        run: sbt scalafmtCheckAll
+
+      - name: Checking import order
+        run: sbt "scalafixAll --check"
+
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Scala
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: adopt@1.11
+
+      - name: Resolving dependencies
+        run: sbt update
+    
+      - name: Compiling production code
+        run: sbt compile
+    
+      - name: Compiling tests
+        run: sbt Test/compile
+    
+      - name: Running unit tests
+        run: sbt test
+    
+      - name: Testing sbt plugin
+        run: sbt scripted
+
+  run-examples:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Scala
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: adopt@1.11
+    
+      - name: Running examples
```

Comment:
Again, worth considering how the overhead of setting up a VM with Scala compares to any gains from parallel execution here